kratos/tests/auxiliar_files_for_python_unittest/cad_json_files/single_square.cad.json

```diff
@@ -0,0 +1,235 @@
+                    "control_points": [
+                      [
+                        11,
+                        [ 0, 10, 0, 1 ]
+                      ],
+                      [
+                        12,
+                        [ 0, 0, 0, 1 ]
+                      ]
+                    ]
+                  }
+                }
+              ]
+            }
+          ],
+          "embedded_loops": [],
+          "embedded_edges": [],
+          "embedded_points": []
+        }
+      ],
+      "edges": [
+        {
+          "brep_id": 4,
+          "3d_curve": {
+            "degree": 1,
+            "knot_vector": [ 0, 0, 15, 15 ],
+            "active_range": [ 0, 15 ],
+            "control_points": [
+              [
+                13,
+                [ 0, 10, 0, 1 ]
+              ],
+              [
+                14,
+                [ 15, 10, 0, 1 ]
+              ]
+            ]
+          },
+          "topology": [
+            {
+              "brep_id": 1,
+              "trim_index": 0,
+              "relative_direction": true
+            }
+          ]
+        },
+        {
+          "brep_id": 2,
+          "3d_curve": {
+            "degree": 1,
+            "knot_vector": [ 0, 0, 10, 10 ],
+            "active_range": [ 0, 10 ],
+            "control_points": [
+              [
+                15,
+                [ 15, 10, 0, 1 ]
+              ],
+              [
+                16,
+                [ 15, 0, 0, 1 ]
+              ]
+            ]
+          },
+          "topology": [
+            {
+              "brep_id": 1,
+              "trim_index": 1,
+              "relative_direction": true
+            }
+          ]
+        },
+        {
+          "brep_id": 5,
+          "3d_curve": {
+            "degree": 1,
+            "knot_vector": [ -15, -15, 0, 0 ],
+            "active_range": [ -15, 0 ],
+            "control_points": [
+              [
+                17,
+                [ 15, 0, 0, 1 ]
+              ],
+              [
+                18,
+                [ 0, 0, 0, 1 ]
+              ]
+            ]
+          },
+          "topology": [
+            {
+              "brep_id": 1,
+              "trim_index": 2,
+              "relative_direction": true
+            }
+          ]
+        },
+        {
+          "brep_id": 6,
+          "3d_curve": {
+            "degree": 1,
+            "knot_vector": [ -10, -10, 0, 0 ],
+            "active_range": [ -10, 0 ],
+            "control_points": [
+              [
+                19,
+                [ 0, 0, 0, 1 ]
+              ],
+              [
+                20,
+                [ 0, 10, 0, 1 ]
+              ]
+            ]
+          },
+          "topology": [
+            {
+              "brep_id": 1,
+              "trim_index": 3,
+              "relative_direction": true
+            }
+          ]
+        }
+      ],
+      "vertices": []
+    }
+  ]
+}
```

Comment:
I think the part of Brep in this format is well crafted. So I would start thinking about how to add normal geometries to it in case we want to have a model geometry file which has non-nurbs/brep entities. (Like reading an STL or IGES and writing our model geometry file for later use)

Reply:
I think that for this we will need another utility, which is completely independent from this one. A IgesInput, and more...

I tried to summarize in  #6343 the stages that came into my mind. I think we should define clearly the stages in the analysis_stage. And even if we don't add an additional function we should clearify exactly what should happen - when.